docs/madr/decisions/056-reachable-svc-mesh-service.md

```diff
@@ -0,0 +1,239 @@
+# Reachable and Autoreachable services with MeshService and MeshExternalService
+
+* Status: accepted
+
+## Context and Problem Statement
+
+Reachable and autoreachable services allow users to define which services they want to communicate with, reducing the memory footprint of a sidecar. Before the introduction of `MeshService` and `MeshExternalService`, a service wasn't a real object but a value of the tag `kuma.io/service`. Since the introduction of `MeshService` and `MeshExternalService`, that changed, and we can refer to them as real objects. Unfortunately, it doesn't work with the current logic and requires implementation.
+
+### Reachable Services
+The functionality allows the user to define a static list of services they want to communicate with. Currently, on Kubernetes, this is possible by setting the annotation `kuma.io/transparent-proxying-reachable-services` with the values of kuma.io/service
+
+Example:
+
+```yaml
+...
+  annotations:
+    kuma.io/transparent-proxying-reachable-services: ["demo-app_kuma-demo_svc_5000", "redis_kuma-demo_svc_6379"]
```

Comment:
nit: it's `kuma.io/transparent-proxying-reachable-services: "demo-app_kuma-demo_svc_5000,redis_kuma-demo_svc_6379"`

it's a comma separated string. Not that it matters for the rest of madr